### PR TITLE
Implement Rect::intersectsCircle

### PR DESCRIPTION
**Pull request by giginet**
I implement Rect::intersectsCircle.

This method makes enable to check the intersect status between the rect and a circle.

``` cpp
cocos2d::Rect rect = Rect(0, 0, 100, 100);
cocos2d::Vec2 center = Vec2(20, 20);
float radius = 10;
rect.intersectsCircle(center, radius); // => true
```

### CODE REVIEW

File: cocos/math/CCGeometry.cpp [hunk 1]
```diff
@@ -188,6 +188,41 @@ bool Rect::intersectsRect(const Rect& rect) const
              rect.getMaxY() <      getMinY());
 }
 
+bool Rect::intersectsCircle(const cocos2d::Vec2 &center, float radius) const
+{
+    Vec2 rectangleCenter = Vec2((origin.x + size.width / 2),
+                                (origin.y + size.height / 2));
+
```
Comment (minggo) [inline cocos/math/CCGeometry.cpp:195]: Use `Vec2 rectangleCenter((origin.x + size.width / 2), (origin.y + size.height / 2))` to optimize it?

File: cocos/math/CCGeometry.cpp [hunk 2]
```diff
@@ -188,6 +188,41 @@ bool Rect::intersectsRect(const Rect& rect) const
...
+        return true;
+    }
+    
+    if (circleDistance.y <= (h))
+    {
+        return true;
+    }
+    
+    float cornerDistanceSq = powf(circleDistance.x - w, 2) +
+    pow(circleDistance.y - h, 2);
```
Comment (minggo) [inline cocos/math/CCGeometry.cpp]: Why one use `powf` and another one use `pow`?
How about writing in one line?

File: cocos/math/CCGeometry.cpp [hunk 3]
```diff
@@ -188,6 +188,41 @@ bool Rect::intersectsRect(const Rect& rect) const
...
+    
+    return (cornerDistanceSq <= (pow(radius, 2)));
```
Comment (minggo) [inline cocos/math/CCGeometry.cpp]: Can not quite understand it.

**Comment by giginet** [comment]
@minggo Thank you for your reviewing.

I fixed some issues.

This method is based on this logic. http://stackoverflow.com/a/402010/1673828

**Pull request merged**